scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/fgboost/FGBoostServiceImpl.scala

```diff
@@ -0,0 +1,171 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.ppml.fgboost
+
+import com.intel.analytics.bigdl.ppml.base.DataHolder
+import com.intel.analytics.bigdl.ppml.common.FLPhase
+import com.intel.analytics.bigdl.ppml.generated.FGBoostServiceProto._
+import com.intel.analytics.bigdl.ppml.generated.{FGBoostServiceGrpc, FGBoostServiceProto}
+import com.intel.analytics.bigdl.ppml.vfl.fgboost.VflGBoostAggregator
+import io.grpc.stub.StreamObserver
+
+class FGBoostServiceImpl(clientNum: Int) extends FGBoostServiceGrpc.FGBoostServiceImplBase{
+  val aggregator = new VflGBoostAggregator()
+  aggregator.setClientNum(clientNum)
+  override def downloadTable(request: DownloadTableRequest,
+                             responseObserver: StreamObserver[DownloadResponse]): Unit = {
+    val version = request.getMetaData.getVersion
+    val data = aggregator.getServerData(FLPhase.LABEL).getTableStorage().serverData
+    if (data == null) {
+      val response = "Your required data doesn't exist"
+      responseObserver.onNext(DownloadResponse.newBuilder.setResponse(response).setCode(0).build)
+      responseObserver.onCompleted()
+    }
+    else {
+      val response = "Download data successfully"
+      responseObserver.onNext(
+        DownloadResponse.newBuilder.setResponse(response).setData(data).setCode(1).build)
+      responseObserver.onCompleted()
+    }
+  }
+
+  override def uploadTable(request: UploadTableRequest,
+                           responseObserver: StreamObserver[UploadResponse]): Unit = {
+    val clientUUID = request.getClientuuid
+    val data = request.getData
+    val version = data.getMetaData.getVersion
+    try {
+      aggregator.putClientData(FLPhase.LABEL, clientUUID, version, new DataHolder(data))
+      val response = s"Table uploaded to server at clientID: $clientUUID, version: $version"
+      responseObserver.onNext(UploadResponse.newBuilder.setResponse(response).setCode(0).build)
+    } catch {
+      case e: Exception =>
+        val response = UploadResponse.newBuilder.setResponse(e.getMessage).setCode(1).build
+        responseObserver.onNext(response)
+        responseObserver.onCompleted()
+    } finally {
+
+    }
+
+  }
+
+  override def split(request: SplitRequest, responseObserver: StreamObserver[SplitResponse]): Unit = {
+    val clientUUID = request.getClientuuid
+    val split = request.getSplit
+
+    val version = -1 // version is not needed in fgboost
+    try {
+      aggregator.putClientData(FLPhase.SPLIT, clientUUID, version, new DataHolder(split))
+      val bestSplit = aggregator.getBestSplit()
```

Comment:
Is this correct? `putClientData` above will call `aggregateSplit`, which will compute best split and clear `SplitStorage`

Reply:
Logic issues would be solved in functionality PR